react/src/filter/Filters.tsx

```diff
@@ -13,29 +17,41 @@ export function Filters({ canvasData }: { canvasData: GetCanvasEmbedResponse })
 
     return (
         <section>
-            {filtersVisible?.map((filter) => (
-                <div key={filter.filterId} className="flex gap-3">
-                    <MultiSelectInput
-                        value={selectedFilters[filter.variable]}
-                        onChange={(item: string) => {
-                            if (item === '' || item == null) {
-                                updateFilter({});
-                                return;
-                            }
-                            const variable = filter.variable;
-                            updateFilter({ [variable]: item });
-                        }}
-                        defaultOption="Select Filter"
-                        // @ts-ignore
-                        options={canvasData.filters.uniqueValues[filter.filterType.storeId]}
-                    />
-                    {valueIsSelected && (
-                        <button onClick={() => updateFilter({})} className="text-xs text-blue-700">
-                            Clear Filter
-                        </button>
-                    )}
-                </div>
-            ))}
+            {filtersVisible?.map((filter) => {
+                let filterOptions;
+
+                if (filterOptionsCache.has(filter.filterId)) {
+                    filterOptions = filterOptionsCache.get(filter.filterId);
+                } else {
+                    // @ts-ignore
```

Comment:
Can we kill `ts-ignore` and add safety checks?

Reply:
Not sure why Rust isn't generating the types for `storeId`, let's discuss offline

Reply:
Ok updated and created new Filter component